utils/dockerd/Config.in

```diff
@@ -1,95 +1,83 @@
-config DOCKER_KERNEL_OPTIONS
-	bool "Enable Basic kernel support for Docker"
-	depends on PACKAGE_docker-ce
-	default y
-	select KERNEL_CGROUPS
-	select KERNEL_CGROUP_CPUACCT
+# These options are mostly specified by https://github.com/moby/moby/blob/master/contrib/check-config.sh
+
+config DOCKER_CGROUP_OPTIONS
+	bool "Enable available kernel support for CGroupsV1"
+	default n
+	depends on PACKAGE_dockerd
 	select KERNEL_CGROUP_DEVICE
 	select KERNEL_CGROUP_FREEZER
-	select KERNEL_CGROUP_SCHED
-	select KERNEL_NAMESPACES
-	select KERNEL_CPUSETS
-	select KERNEL_MEMCG
-	select KERNEL_KEYS
-	select KERNEL_POSIX_MQUEUE
+	select KERNEL_NET_CLS_CGROUP
+	select KERNEL_CGROUP_NET_PRIO
+	select PACKAGE_cgroupfs-mount
 	help
-	  Select needed kernel options for Docker. Options include
-	  cgroups, namespaces and other miscellaneous options.
-	  see also https://github.com/docker/engine/blob/master/contrib/check-config.sh
+		Selects kernel options to enable CGroups V1.
 
-config DOCKER_SECCOMP
-	bool "Enable support for seccomp in Docker"
-	depends on PACKAGE_docker-ce
+config DOCKER_GENERALLY_NECESSARY
+	bool "Enable generally necessary kernel support for Docker"
 	default n
-	select KERNEL_SECCOMP
-	select PACKAGE_libseccomp
+	depends on PACKAGE_dockerd
+	select KERNEL_KEYS
```

Comment:
Selecting `CONFIG_KEYS` by default for target kernels with large enough flash space might generally be a good idea.
I've been preparing a cleanup of the kernel keyring options, please see my staging tree:
https://git.openwrt.org/?p=openwrt/staging/dangole.git;a=commit;h=308aee3f80cbb91b2ddae67a208062f36b42c16a

Reply:
Yeah, that sounds pretty good.

Reply:
I've posted those changes to the mailing list, if there are no objections, I'll merge them this week:
https://patchwork.ozlabs.org/project/openwrt/patch/X9U+Vn+oBpAx3H+y@makrotopia.org/

Reply:
Pushed: https://github.com/openwrt/openwrt/commit/23049f9c31f030829980b7e32e424435572ae9ac
Please adapt the PR accordingly, other than that it's ready to get merged.

Reply:
Removed config option from dockerd